language/llama3-405b/dataset.py

```diff
@@ -0,0 +1,129 @@
+import random
+import os
+import time
+import numpy as np
+import torch
+from datasets import load_dataset, load_from_disk
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from torch.nn.functional import pad
+from torch.utils.data import DataLoader
+from typing import Optional, Dict, Sequence
+import io
+
+# import utils
+import copy
+import pickle
+
+import logging
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("Llama-70B-Dataset")
+
+
+class Dataset:
+    def __init__(
+        self,
+        model_name=None,
+        total_sample_count=24576,
+        perf_count_override=None,
+        dataset_path=None,
+        dtype="bfloat16"
+    ):
+        self.model_name = model_name or f"Meta-Llama-3.1-405B-Instruct{'-FP8' if dtype == 'float8' else ''}"
+        self.dataset_path = dataset_path
+        self.max_length = 1024
+
+        # self.total_sample_count = total_sample_count
+
+        self.load_tokenizer()
+        self.load_processed_dataset()
+
+        self.total_sample_count = min(len(self.input_ids), total_sample_count)
+        self.perf_count = perf_count_override or self.total_sample_count
+
+    def load_tokenizer(self):
+        """Returns tokenizer"""
+        pass
+        # self.tokenizer = AutoTokenizer.from_pretrained(
+        #     self.model_name,
+        #     model_max_length=1024,
+        #     padding_side="left",
+        #     use_fast=False,
+        # )
+
+        # self.tokenizer.pad_token = self.tokenizer.eos_token
+
+    def load_processed_dataset(self):
+        if not os.path.isfile(self.dataset_path):
+            log.warn(
+                "Processed pickle file {} not found. Please check that the path is correct".format(
+                    self.dataset_path
+                )
+            )
+
+        print("Loading dataset...")
```

Comment:
We have loggers (log), so probably should use it instead of printing. Apply for all below cases

Reply:
Done